### ISSUE TITLE
Ubuntu 16.04 LTS support

### ISSUE THREAD

**Issue by jendap** [NVIDIA/nvidia-docker#81]
Can we get it please? It is officially released already.

**Comment by flx42** [comment]
Do you mean for the images on the DockerHub?
Or for the nvidia-docker/nvidia-docker-plugin binaries?

**Comment by jendap** [comment]
Both :-) But I hope the nvidia-docker binary works. The images on DockerHub would be nice. There are newer versions of all the packages. It would simplify a lot the images build on top of nvidia docker image.

**Comment by lukeyeager** [comment]
Building and installing on 16.04 worked just fine for me.
```sh
git clone git@github.com:NVIDIA/nvidia-docker.git
cd nvidia-docker/
make
sudo PREFIX=/usr/local/bin make install
sudo nvidia-docker volume setup
nvidia-docker run --rm nvidia/cuda nvidia-smi
```

**Comment by jendap** [comment]
Yes, that works. But there is no docker image with "FROM ubuntu:16.04" uploaded to some docker hub. There is only 14.04 - https://github.com/NVIDIA/nvidia-docker/blob/master/ubuntu-14.04/cuda/7.5/runtime/Dockerfile#L1 and https://hub.docker.com/r/nvidia/cuda/

It is important as my understanding is that we can't build it due to some redistributions restrictions in nvidia license. However when you guys create and upload the image it should all be fine :-)

**Comment by flx42** [comment]
Sorry about the delay! I now have an answer for you: ubuntu 16.04 images will be available as soon as CUDA 8.0 RC is out, and it is due at the end of the month.

**Comment by jendap** [comment]
Is it coming? We would like to start building nighltly builds and next release of tensorflow with support for 16.04. But we need docker support for our build infrastructure.

(even if it would be just older cuda for now)

**Comment by flx42** [comment]
It should be coming today, unless something explodes unexpectedly :)

**Comment by jendap** [comment]
Bring it in!

Will it have only 8.0RC or even older 7.5? There are 3 versions in https://github.com/NVIDIA/nvidia-docker/tree/master/ubuntu-14.04/cuda/

It would be great to have 7.5 (non RC) on 16.04. Today. It should make it to next tensorflow release ;-)

**Comment by flx42** [comment]
8.0 RC only, sorry.
I did a quick test and saw a small compiling problem with TF because CUDA 8.0 unified some data types. The problem was in this patch:
https://github.com/tensorflow/tensorflow/commit/523055469c8a61425e3b8f104be67787c2933ccb

**Comment by jendap** [comment]
Publish the container. We will start using it in [ci_build](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/ci_build) and on [ci.tensorflow.org](ci.tensorflow.org). We will see the error and we can fix it ;)

**Comment by flx42** [comment]
Currently pushing, should soon land with a OS prefix for now: `nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04`

**Comment by flx42** [comment]
Please test :)

**Comment by jendap** [comment]
Great! Thank you!

I will test it tonight.

**Comment by flx42** [comment]
@jendap: were you able to test the images? Does it work for you?
If yes, can I close this issue?

**Comment by jendap** [comment]
Not yet. I had to firefight more important issues. I will look at it this weekend, build and upload tensorflow container build on top of this.  But we can close this. The container seems to be fine. And if not we can always open new issue.

Thank you!

**Comment by tiangolo** [comment]
I'm not sure if this is the right place to ask, but I can't find the CUDA Docker images for Ubuntu 16 or any results of the tests...

Did you upload the new Docker images @flx42? Did your tests worked @jendap?

I really don't know if having a 16.04 host with 14.04 containers would work or not. I'm delaying upgrading to 16.04 waiting for this mainly, but I don't know where else should I check.

**Comment by 3XX0** [comment]
@tiangolo Tags are available here (end of the page):
https://hub.docker.com/r/nvidia/cuda/tags

**Comment by flx42** [comment]
And yes, having a 16.04 host with 14.04 containers should work just fine.

**Comment by jendap** [comment]
And if you ask about tensorflow ... I'm working on it. Soon....

**Comment by tiangolo** [comment]
Wow, that's a responsive team (or multi-team)!

Ah! I get it, thanks @3XX0

And thanks @flx42, I'll upgrade right away then.

Great @jendap. Could you post here again when you finish? That way we (and others in the future) can check it.

**Comment by jendap** [comment]
@3XX0 and @flx42 are very responsive. Too much responsive. They may be bots :-)

**Comment by flx42** [comment]
I wish we had a GitHub answer bot, could we do that with TensorFlow? ;)

**Comment by jendap** [comment]
Of course! It just need more training data. Make another 50 000 comments please! ;)

**Comment by tiangolo** [comment]
LOL you made me laugh (like... aloud)... here's for the training set! :laughing:

**Comment by ra9hur** [comment]
From the link,
https://github.com/NVIDIA/nvidia-docker/blob/master/ubuntu-14.04/cuda/8.0/runtime/cudnn5/Dockerfile

See that, cuda8.0/cudnn5_runtime are available only on Ubuntu 14.04. Would there be a version for Ubuntu 16.04/16.10 shortly ?

**Comment by flx42** [comment]
Locking, please don't bump old issues.
